src/librustc/mir/interpret/mod.rs

```diff
@@ -526,7 +528,7 @@ impl Allocation {
             relocations: Relocations::new(),
             undef_mask: UndefMask::new(size),
             align,
-            runtime_mutability: Mutability::Immutable,
+            mutability: Mutability::Mutable,
```

Comment:
Why is this made mutable?

Reply:
This is how all allocations are created. Previously, mutability was entirely ignored by the miri engine, so this flag only mattered once the static got interned (where it gets set, in `intern_static` in `memory.rs`).

But now that the engine enforces mutability, we have to make *some* allocations mutable or else we have no heap ;)